lib/dependency_validator.dart

```diff
@@ -24,11 +25,16 @@ import 'src/utils.dart';
 export 'src/constants.dart' show commonBinaryPackages;
 
 /// Check for missing, under-promoted, over-promoted, and unused dependencies.
-void run() {
+Future<Null> run() async {
   if (!File('pubspec.yaml').existsSync()) {
     logger.shout('pubspec.yaml not found');
     exit(1);
   }
+  if (!File('.dart_tool/package_config.json').existsSync()) {
+    logger.shout('.dart_tool/package_config.json not found');
+    exit(1);
+  }
```

Comment:
This is effectively a proxy for whether or not `pub get` has been run. In the ticket I was suggesting that we log a warning when this occurs telling the user that some packages may incorrectly marked as unused until a `pub get` is run so that we can do this additional executable checking logic.. but since we're working on a major release, I guess we really just need to decide whether we should make that a requirement. Thoughts?

If we do decide that the existence of the package config file (in other words, having run a `pub get`) is a requirement for running dependency validator, then we should update this error message to be more helpful by telling the user to run `pub get` first.

Reply:
>In the ticket I was suggesting that we log a warning when this occurs telling the user that some packages may incorrectly marked as unused

Sorry, that's totally in the ticket description, I didn't read the ticket closely enough :coneofshame:.

>I guess we really just need to decide whether we should make that a requirement. Thoughts?

I'd lean towards requiring a `pub get`. Personally I'd prefer a tool that is persnickety but always gives the same result, to one that gives different answers depending on circumstances.

>If we do decide that the existence of the package config file (in other words, having run a pub get) is a requirement for running dependency validator, then we should update this error message to be more helpful by telling the user to run pub get first.

Roger :salute:.

Reply:
I'll hold off on any changes related to this until I hear back from you.

Reply:
Yeah I agree, better to be consistent and a `pub get` is a reasonable bare minimum for running a check on a dart package IMO. I dropped a message in our team chat to see if anyone else has thoughts.